python/lsst/ts/externalscripts/maintel/setup_whitelight_flats.py

```diff
@@ -0,0 +1,158 @@
+              sequence_name:
+                description: Name of sequence in MTCalsys
+                type: string
+                default: whitelight_r
+
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def set_metadata(self, metadata):
+        metadata.duration = 30
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : ``self.cmd_configure.DataType``
+
+        """
+        self.log.info("Configure started")
+        if self.mtcalsys is None:
+            self.log.debug("Creating MTCalSys.")
+            self.mtcalsys = MTCalsys(domain=self.domain, log=self.log)
+            await self.mtcalsys.start_task
+
+        self.sequence_name = config.sequence_name
+        self.mtcalsys.load_calibration_config_file()
+        self.mtcalsys.assert_valid_configuration_option(name=self.sequence_name)
+
+        self.config_data = self.mtcalsys.get_calibration_configuration(
+            self.sequence_name
+        )
+
+        self.electrometer = self.mtcalsys.electrometer
+        self.fiberspec_red = self.mtcalsys.fiberspectrograph_red
+        self.fiberspec_blue = self.mtcalsys.fiberspectrograph_blue
+        self.linearstage_led_focus = self.mtcalsys.linearstage_led_focus
+        self.linearstage_led_select = self.mtcalsys.linearstage_led_select
+        self.linearstage_projector_select = self.mtcalsys.linearstage_projector_select
+        self.led_projector = self.mtcalsys.rem.ledprojector
+
+        self.log.info("Configure completed")
+
+    async def run(self):
+        """Run script."""
+        await self.assert_components_enabled()
+
+        await self.checkpoint("Setting up Calibration Projector")
+        await self.mtcalsys.setup_calsys(sequence_name=self.sequence_name)
+
+        await self.checkpoint("Preparing for Flats")
+        await self.mtcalsys.prepare_for_flat(sequence_name=self.sequence_name)
+
+        # # TO-DO: DM-49065 for mtcalsys.py
+        # params = await self.mtcalsys.get_projector_setup()
+
+        # self.log.info(
+        #     f"Laser Configuration is {params[0]}, \n"
+        #     f"wavelength is {params[1]}, \n"
+        #     f"Interlock is {params[2]}, \n"
+        #     f"Burst mode is {params[3]}, \n"
+        #     f"Cont. mode is {params[4]}"
+        # )
+
+    async def assert_components_enabled(self):
+        """Checks if LEDProjector, Electrometer, Fiber Spectrographs,
+        and all LinearStages are ENABLED
+
+        Raises
+        ------
+        RunTimeError:
+            If either component is not ENABLED"""
+
+        comps = [
+            self.electrometer,
+            self.fiberspec_red,
+            self.fiberspec_blue,
+            self.linearstage_led_focus,
+            self.linearstage_led_select,
+            self.linearstage_projector_select,
+            self.led_projector,
+        ]
+
+        for comp in comps:
```

Comment:
why not do `await self.atcalsys.assert_all_enabled()`?

Reply:
I didn't know that was possible! I hope it still works even though the Laser isn't used on this.

Reply:
It doesn't look like that is an option in mtcalsys.py or base_calsys.py

Reply:
I can't use `assert_all_enabled` because I don't want the TunableLaser to be Enabled

Reply:
Ah okay, in that case you can add the TunableLaser (and anything else in the MTCalSys CSC group you want to ignore) to a list of components to ignore and include the `disable_checks_for_components()` in your config method. 

Here's an example of a script using a configurable parameter to ignore some components of the TCS if needed. You could add that, and include the TunableLaser by default. The nice thing about doing it this way is if you ever need to run the script and ignore other CSCs you can pass those through the script config and will not need to update the script itself.